src/GiraffeGenerator.Core/AST.fs

```diff
@@ -290,8 +297,19 @@ let app funExpr argExpr =
 let appI funExpr argExpr =
     SynExpr.App(ExprAtomicFlag.NonAtomic, true, funExpr, argExpr, r)
 
+/// Expression for type application
+/// {a}<{b[0],b[1]...}>
+let typeApp a b =
+    SynExpr.TypeApp(a,r,b,[],None,r,r)
```

Comment:
Not really, because `typeApp` generates expression of type application (e.g. `something.CallGeneric<A,B,C>`) while `genericType` generates, well, generic type (e.g. `GenericType<A,B,C>`)